src/types/web-resource.ts

```diff
@@ -0,0 +1,140 @@
+import * as TypeUtils from '../type-utils';
+
+export type WebResource = {
+	filename: string;
+	href: string;
+	contentType?: string;
+	contentDisposition?: string;
+	size?: number;
+};
+
+export const types = {
+	postgres: 'JSONB',
+	mysql: 'JSON',
+	websql: 'TEXT',
+	odata: {
+		name: 'Self.WebResource',
+		complexType: `\
+<ComplexType Name="WebResource">
+	<Property Name="filename" Nullable="false" Type="Edm.String"/>\
+	<Property Name="href" Nullable="false" Type="Edm.String"/>\
+	<Property Name="contentType" Nullable="true" Type="Edm.String"/>\
+	<Property Name="contentDisposition" Nullable="true" Type="Edm.String"/>\
+	<Property Name="size" Nullable="true" Type="Edm.Int64"/>\
+</ComplexType>`,
+	},
+};
+
+export const nativeProperties = {
+	has: {
+		Filename: (stringifiedResource: string) => [
+			'ExtractJSONPathAsText',
+			stringifiedResource,
+			['TextArray', ['EmbeddedText', 'filename']],
+		],
+		HRef: (stringifiedResource: string) => [
+			'ExtractJSONPathAsText',
+			stringifiedResource,
+			['TextArray', ['EmbeddedText', 'href']],
+		],
+		'Content Type': (stringifiedResource: string) => [
+			'ExtractJSONPathAsText',
+			stringifiedResource,
+			['TextArray', ['EmbeddedText', 'contentType']],
+		],
+		'Content Disposition': (stringifiedResource: string) => [
+			'ExtractJSONPathAsText',
+			stringifiedResource,
+			['TextArray', ['EmbeddedText', 'contentDisposition']],
+		],
+		Size: (stringifiedResource: string) => [
```

Comment:
It might be worth using the abstract-sql-compile for these to type-check them, eg
```suggestion
		Size: (stringifiedResource: string): CastNode => [
```

Reply:
done!